src/fetcher.ts

```diff
@@ -27,6 +29,7 @@ export const withDefaults = (request: Partial<IHttpRequest>) => ({
 });
 
 export function fetcher(opts: IClientConfiguration = {}) {
+  const fetchRequest = fetchRequestFactory(opts.fetch || global.fetch);
```

Comment:
We can't use `global` since this only exists in NodeJS environment.

Let's move the default value to the factory.

```suggestion
  const fetchRequest = fetchRequestFactory(opts.fetch);
```

Reply:
`global` is available in the browser through module bundler like `webpack` and others so this is safe but agree it's not necessary.

Reply:
Right but no everyone uses Webpack,

Also, you could use `globalThis` instead since that is now part of the JS spec.